x-pack/plugins/security_solution/server/lib/siem_migrations/rules/api/rules/helpers.ts

```diff
@@ -0,0 +1,186 @@
+    );
+    if (installedRule) {
+      rulesToUpdate.push({
+        ...ruleToInstall,
+        elastic_rule: {
+          ...ruleToInstall.elastic_rule,
+          id: installedRule.id,
+        },
+      });
+      return;
+    }
+
+    // If prebuilt rule is not installed, then keep reference to install it
+    const installableRule = installableRules.find(
+      (rule) => rule.rule_id === ruleToInstall.elastic_rule.prebuilt_rule_id
+    );
+    if (installableRule) {
+      assetsToInstall.push(installableRule);
+    }
+  });
+
+  // Filter out any duplicates which can occur when multiple translated rules matched the same prebuilt rule
+  const filteredAssetsToInstall = assetsToInstall.filter(
+    (value, index, self) => index === self.findIndex((rule) => rule.rule_id === value.rule_id)
+  );
+
+  // TODO: we need to do an error handling which can happen during the rule installation
+  const { results: installedRules } = await createPrebuiltRules(
+    detectionRulesClient,
+    filteredAssetsToInstall
+  );
+  await performTimelinesInstallation(securitySolutionContext);
+
+  installedRules.forEach((installedRule) => {
+    const rules = rulesToInstall.filter(
+      (rule) => rule.elastic_rule.prebuilt_rule_id === installedRule.result.rule_id
+    );
+    rules.forEach((prebuiltRule) => {
+      rulesToUpdate.push({
+        ...prebuiltRule,
+        elastic_rule: {
+          ...prebuiltRule.elastic_rule,
+          id: installedRule.result.id,
+        },
+      });
+    });
+  });
+
+  return rulesToUpdate;
+};
+
+const installCustomRules = async (
+  rulesToInstall: RuleMigrationToInstall[],
+  detectionRulesClient: IDetectionRulesClient,
+  logger: Logger
+): Promise<UpdateRuleMigrationInput[]> => {
+  const rulesToUpdate: UpdateRuleMigrationInput[] = [];
+  await Promise.all(
+    rulesToInstall.map(async (rule) => {
+      if (!rule.elastic_rule.query || !rule.elastic_rule.description) {
+        return;
+      }
+      try {
+        const payloadRule: RuleCreateProps = {
+          type: 'esql',
+          language: 'esql',
+          query: rule.elastic_rule.query,
+          name: rule.elastic_rule.title,
+          description: rule.elastic_rule.description,
+          severity: DEFAULT_TRANSLATION_SEVERITY,
+          risk_score: DEFAULT_TRANSLATION_RISK_SCORE,
+        };
+        const createdRule = await detectionRulesClient.createCustomRule({
+          params: payloadRule,
+        });
+        rulesToUpdate.push({
+          ...rule,
+          elastic_rule: {
+            ...rule.elastic_rule,
+            id: createdRule.id,
+          },
+        });
+      } catch (err) {
+        // TODO: we need to do an error handling which can happen during the rule creation
+        logger.debug(`Could not create a rule because of error: ${JSON.stringify(err)}`);
+      }
+    })
+  );
+  return rulesToUpdate;
+};
+
+export const installRules = async (
```

Comment:
All the logic here is related to rules installation (which is perfect 👍) however the name of the file does not reflect that. We already have a _rules/api/util_ directory, would it make sense to move this file to _rules/api/util/installation.ts_?

Reply:
done